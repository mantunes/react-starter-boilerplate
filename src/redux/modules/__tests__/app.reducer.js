import configureMockStore from 'redux-mock-store';

import AppReducer, { addName, removeName } from '../app';

const mockStore = configureMockStore([]);
const store = mockStore({});

beforeEach(() => {
  store.clearActions();
});

it('INITIAL STATE', () => {
  const action = { type: 'DUMMY' };
  expect(AppReducer(undefined, action)).toMatchSnapshot();
});

it('ADD_NAME (Rui Miguel)', () => {
  expect(AppReducer(undefined, addName('Rui Miguel'))).toMatchSnapshot();
});

it('SUBTRACT_NAME (Rui Miguel)', () => {
  expect(AppReducer(undefined, removeName(0))).toMatchSnapshot();
});
