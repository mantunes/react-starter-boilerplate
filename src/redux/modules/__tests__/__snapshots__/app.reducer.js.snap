// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_NAME (Rui Miguel) 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
    Object {
      "id": 3,
      "name": "Rui",
    },
    Object {
      "id": 1506534848622,
      "name": "Rui Miguel",
    },
  ],
  "post": null,
}
`;

exports[`INITIAL STATE 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
    Object {
      "id": 3,
      "name": "Rui",
    },
  ],
  "post": null,
}
`;

exports[`SUBTRACT_NAME (Rui Miguel) 1`] = `
Object {
  "list": Array [
    Object {
      "id": 1,
      "name": "Joao",
    },
    Object {
      "id": 2,
      "name": "Sara",
    },
    Object {
      "id": 3,
      "name": "Rui",
    },
  ],
  "post": null,
}
`;
